org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/BuildManager.xtend

```diff
@@ -36,11 +36,15 @@ class BuildManager {
 
     val dirtyFiles = <URI>newLinkedHashSet
     val deletedFiles = <URI>newLinkedHashSet
+    
+    def Buildable submit(List<URI> dirtyFiles, List<URI> deletedFiles) {
+    	queue(this.dirtyFiles, deletedFiles, dirtyFiles)
+        queue(this.deletedFiles, dirtyFiles, deletedFiles)
+        return [cancelIndicator|internalBuild(cancelIndicator)]
```

Comment:
what happens with the changes if the build got canceled? they need to be queued again.